src/app/components/signing-dialog.component.ts

```diff
@@ -234,18 +254,67 @@ export class SigningDialogComponent {
     return summary.join('\n');
   }
 
-  signWithExtension() {
-    this.sign.emit({ signed: true, key: 'extension' });
+  async signWithExtension() {
+    this.validationError = '';
+    
+    try {
+      // Check if extension is available
+      if (!window.nostr) {
+        this.validationError = 'Nostr extension not found. Please install a Nostr extension.';
+        return;
+      }
+
+      // Get public key from extension
+      const extensionPubkey = await window.nostr.getPublicKey();
```

Comment:
It's better to just ask to sign it and check the pubkey on returned value, then user doesn't have to respond to two prompts.